dockerfile written for building docker image

TheMohit2003 opened the pull request:
# PR description 
- This pr addresses #360 

## How to test   

1. Ensure Docker is installed on your system.
2. Navigate to the root directory of the project and run:
    ```bash
    docker build -t ruxalab .
    ```
3. Once the image is built, run it using:
    ```bash
    docker run -d --env-file .env -p 5000:5000 ruxalab
    ```
4. Visit `http://localhost:5000` in your browser to view the application.

TheMohit2003 commented:
@jvJUCA , i have written and tested the docker file . But , haven't update the readme with docker setup steps. And coming to env handling in the project , i have used `--env-file` flag to inject the `env` keys directly in the running container. I will soon send the updated readme on this branch itself .

asr2003 commented on dockerfile:
```diff
@@ -0,0 +1,28 @@
+
+FROM node:16 AS build-stage
```
Please use `node:lts` instead node:16 to mitigate risks of compatibility in future

KarinePistili replied:
Good suggestion!

asr2003 commented on dockerfile:
```diff
@@ -0,0 +1,28 @@
+RUN npm run build
+
+# Serve the app using Node.js
+FROM node:16 AS production-stage
```
Use` node:alpine` to reduce the size of final image

KarinePistili replied:
Hello @asr2003 I think this are good suggestions!

KarinePistili replied:
I think it would good to use node latest here as well

asr2003 replied:
Yes, we can use node lts for build stage. node-alpine is the lighter image which we can use safely to run the already built files in Stage-1. It's the mantra we follow to reduce docker image :)

KarinePistili replied:
Cool!

asr2003 reviewed (commented):
I would suggest having two docker files for prod and dev environments are helpful rather than a single Dockerfile.

KarinePistili commented:
@xemyst were you able to test it on your environment as well?

MurilonND commented:
Hello @TheMohit2003, I just finished testing your commit, and for me, the code isn't working (tested after merging with the develop so this could be why), so I just changed the code and got it working.

![Screenshot from 2024-04-15 15-03-16](https://github.com/ruxailab/RUXAILAB/assets/81724197/b2f421a3-8a38-43cb-97c4-ffbbb4b643f5)

MurilonND reviewed (changes requested):
Hello @TheMohit2003, I just finished testing your commit, and for me, the code isn't working (tested after merging with the develop so this could be why), so I just changed the code and got it working.

![Screenshot from 2024-04-15 15-03-16](https://github.com/ruxailab/RUXAILAB/assets/81724197/b2f421a3-8a38-43cb-97c4-ffbbb4b643f5)

KarinePistili reviewed (changes requested):
Please review the changes provided by the reviewers and change accordingly

TheMohit2003 commented:
@MurilonND , can you discuss the changes ?

MurilonND commented:
Sure, what you want to know?

TheMohit2003 commented:
> Sure, what you want to know?

What error did you get, and what file changes did you make ?

MurilonND commented:
Very simple, just changed the node version to alpine (optimization), and the ERROR that I got was:

[build-stage 6/6] RUN npm run build:
0.364 npm ERR! Missing script: "build"
0.364 npm ERR! 
0.364 npm ERR! To see a list of scripts, run:
0.365 npm ERR!   npm run
0.365 
0.365 npm ERR! A complete log of this run can be found in:
0.366 npm ERR!     /root/.npm/_logs/2024-04-15T17_30_45_835Z-debug-0.log
------
dockerfile:12
--------------------
  10 |     COPY . .
  11 |
  12 | >>> RUN npm run build
  13 |
  14 |     # Serve the app using Node.js
--------------------
ERROR: failed to solve: process "/bin/sh -c npm run build" did not complete successfully: exit code: 1

The develop doesn't have a "build" script, instead, it has "build-dev" or "build-prod."

KarinePistili commented:
@TheMohit2003 will you be able to do the changes?

TheMohit2003 commented:
@KarinePistili , My apologies for being late. I will send the commits by tomorrow. Just went through distro hopping ( linux )and need to setup stuff again.

KarinePistili commented:
Okay, thanks! @TheMohit2003

TheMohit2003 commented:
**Updated Dockerfile:**
- Changed the Node.js version in the **build stage** to `node:lts` to ensure long-term support and compatibility.
- Switched to `node:alpine` in the **production stage** to reduce the final image size, improving deployment efficiency.
@KarinePistili

TheMohit2003 commented:
@KarinePistili , should I add docker image setup docs in readme ?

KarinePistili commented:
it would be nice @TheMohit2003 , thanks

TheMohit2003 commented:
@KarinePistili, all tasks on my end are completed. If there are any further changes you need, please let me know. Otherwise, you are free to merge the PR when you are ready.

asr2003 commented:
I will test your changes and provide any changes needed

TheMohit2003 commented:
@asr2003 , did you reviewed the changes ?

JulioManoel commented:
I think it's interesting to put a dockerignore for common-models

TheMohit2003 commented:
@JulioManoel worked for me . Let me check again. Did you create a .env file with all the secrets before running the container?

JulioManoel commented:
Wouldn't it be better to separate two dockerfiles, 1 for develop and another for production?

JulioManoel commented:
@TheMohit2003 there was an error when building the container

JulioManoel commented:
I have the .env, I gave npm i to the project and after that I tried to build docker using the README command:

```bash
docker build -t uxremotelab .
```

TheMohit2003 commented:
@JulioManoel , I can do it that way, but the assigned reviewers haven't said anything regarding the current changes . Let me take their reviews in consideration, and then I will write a separate dev and production based dockerfile . @KarinePistili @asr2003 .

JulioManoel commented:
Sorry it was an internet problem ksks 😅

KarinePistili commented:
Thank you for revision @JulioManoel @MurilonND @asr2003 and @xemyst
